modules/standard/Curl.chpl

```diff
@@ -17,24 +17,326 @@
  * limitations under the License.
  */
 
-/* Libcurl support 
+/*
+
+This module provides support for libcurl, enabling Chapel programs
+to work with many network protocols. 
+
+The `curl homepage <http://curl.haxx.se/libcurl/>`_ describes libcurl thus::
+
+ libcurl is a free and easy-to-use client-side URL transfer library, supporting
+ DICT, FILE, FTP, FTPS, Gopher, HTTP, HTTPS, IMAP, IMAPS, LDAP, LDAPS, POP3,
+ POP3S, RTMP, RTSP, SCP, SFTP, SMTP, SMTPS, Telnet and TFTP.  libcurl supports
+ SSL certificates, HTTP POST, HTTP PUT, FTP uploading, HTTP form based upload,
+ proxies, cookies, user+password authentication (Basic, Digest, NTLM,
+ Negotiate, Kerberos), file transfer resume, http proxy tunneling and more! 
+
+Dependencies
+------------
+
+The Curl functionality in Chapel is dependent on libcurl. For information on
+how to install libcurl, see the
+`curl installation instructions <http://curl.haxx.se/docs/install.html>`_
+
+The environment variables CHPL_AUXIO_INCLUDE and CHPL_AUXIO_LIBS must be set to
+point to the include and lib directories for libcurl respectively. More
+information on these variables can be found in README.auxIO
+
+.. note::
+
+  If libcurl is installed system-wide you should not need to set these
+  variables.
+
+
+Enabling Curl Support
+---------------------
+
+Once you have ensured that libcurl is installed, and have the two variables
+above defined, set the environment variable CHPL_AUX_FILESYS to 'curl' to enable
+Curl support:
+
+.. code-block:: sh
+
+  export CHPL_AUX_FILESYS=curl
+
+Then, rebuild Chapel by executing 'make' from $CHPL_HOME:
+
+.. code-block:: sh
+
+  make
+
+.. note::
+
+  If Curl support is not enabled (which is the default), all features
+  described below will compile successfully but will result in an error at
+  runtime, saying: "No Curl Support".
+
+For information on how to enable and use Curl while also using other auxiliary
+IO extensions, as well as how to setup the CHPL_AUXIO_INCLUDE and
+CHPL_AUXIO_LIBS environment variables see README.auxIO in this directory.
```

Comment:
I think "this directory" should be replaced with "the technotes folder in the release" or something to that effect.